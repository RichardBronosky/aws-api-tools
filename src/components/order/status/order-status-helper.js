import { getOrderList } from '@services/order-status'
import { announce } from '@helpers/aria-announce'
import { validatePhone } from '@helpers/string-helper'

export const getRelatedOrders = (e, orderId, phoneNumber, loading, setState, setSearchState, ref) => {
  e.preventDefault()
  if (!loading) {
    setSearchState({ loading: true })
    const validPhone = validatePhone(phoneNumber)
    if (validPhone) {
      getOrderList({ orderId: orderId, phoneNumber: phoneNumber.replace(/[^0-9]/g, '') })
        .then(orderList => {
          setState({ orderList: orderList })
          ref && window.scrollTo(0, ref.current.clientHeight)
          setSearchState({ err: null, loading: false })
        })
        .catch(err => {
          const error =
            err && err.message && err.message.data && err.message.data.error && err.response.data.error.message
              ? err.response.data.error.message
              : 'We are unable to retrieve your order, please try again later.'
          announce(error)
          setState({ orderList: {} })
          setSearchState({ err: error, loading: false })
        })
    } else {
      const error = 'Invalid phone number.'
      announce(error)
      setState({ orderList: {} })
      setSearchState({ err: error, loading: false })
    }
  }
}

export const viewOrderList = setState => {
  setState({ orderDetails: null })
}

export const viewOrderDetails = (order, setState) => {
  setState({ orderDetails: order })
  window.scrollTo(0, 0)
}

export const getFinDepositTotal = deposits => {
  let total = 0
  if (deposits.length > 0) {
    for (let i = 0, n = deposits.length; i < n; i++) {
      total += deposits[i].AuthorizedAmount
    }
  }
  return total
}

export const getOtherPayments = order => {
  let otherPayments = []
  if (order) {
    if (order.Deposits && order.Deposits.Deposit.length > 0) {
      otherPayments = [...otherPayments, ...order.Deposits.Deposit]
    }
    if (order.GCDeposits && order.GCDeposits.Deposit.length > 0) {
      otherPayments = [...otherPayments, ...order.GCDeposits.Deposit]
    }
  }
  return otherPayments
}

export const getPaymentType = dep => {
  switch (dep.CashCode) {
    case 'MC':
      return 'Master Card'
    case 'VA':
      return 'Visa'
    case 'DV':
      return 'Discover'
    case 'AMX':
      return 'American Express'
    case 'GC':
      return 'Gift Card'
    case 'PAL':
      return 'PayPal'
    case 'AFF':
      return 'Affirm'
  }
}
