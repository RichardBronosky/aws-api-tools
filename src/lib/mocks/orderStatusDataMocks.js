export const orderStatus = {
  AbleToReschedule: false,
  AdditionalDirections: '**** ALL DAY DELIVERY**** ****NO TIME FRAMES****',
  AffiliateData: '',
  Balance: '0.00',
  BillingAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    State: 'GA',
    Zip: '30328',
  },
  Customer: {
    AllowText2AltPhoneNumber: false,
    AllowText2PhoneNumber: false,
    AltPhoneNumber: '',
    CMKey: '01*202/292-8491',
    CustomerID: 'SE006284958',
    Division: 'SE',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    PhoneNumber: '202/292-8491',
  },
  DeliveryAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    Plus4: '',
    State: 'GA',
    Zip: '30328',
  },
  DeliveryCharge: '119.99',
  DeliveryDate: '2019-08-06T00:00:00.000Z',
  DeliveryType: 'E',
  Deposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '37.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'VA',
      },
    ],
  },
  Division: 'SE',
  EmailAddress: 'jamiamorrow@gmail.com',
  EstimatedDeliveryTimeFrame: '',
  FinDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '600.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'RGE',
      },
    ],
  },
  FinanceAccountNumber: '6019191263030424',
  FinancePlan: '90015-0121',
  FraudFlag: '',
  GCDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '1.95',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'GC',
      },
    ],
  },
  I95ind: false,
  LineItems: {
    LineItem: [
      {
        ChildItems: {
          LineItem: [
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36111639',
              UnitPrice: '193.24',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36211631',
              UnitPrice: '250.76',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
          ],
        },
        PackagePrice: '444',
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '3611163P',
        UnitPrice: '0',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Louie Gray Twin Loft Bed with Desk and Bookcase',
      },
      {
        ChildItems: null,
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '38096613',
        UnitPrice: '29',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Mattie Gray Chevron Bean Bag Chair',
      },
    ],
  },
  OrderDate: '2019-08-04T23:17:03.000Z',
  OrderMessage: 'OK',
  OrderNumber: '14000136I',
  OrderTotal: '638.95',
  PickUpData: {
    Phone: '',
    PickupAddress: { Address1: '', Address2: '', City: '', State: '', Zip: '' },
    PickupHours: { string: null },
    StoreName: '',
    StoreNumber: '980',
  },
  RemoteDeliveryDate: '0001-01-01T00:00:00.000Z',
  RemoteEstimatedDeliveryTimeFrame: '',
  ShipDate: '',
  Status: 'Open',
  StoreNumber: '980',
  TaxAmount: '45.96',
  TaxAreaId: '111211811',
  Active: true,
  DispatchTrack: 'https://fgo-rtg.dispatchtrack.com/track_order/f0d44a2/SE14000136I',
}

export const orderStatusList = [orderStatus]
