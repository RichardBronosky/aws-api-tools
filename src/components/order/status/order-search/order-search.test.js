import React from 'react'
import renderer from 'react-test-renderer'
import OrderSearch from './order-search'
import { orderStatus } from '@mocks/orderStatusDataMocks'
import * as orderStatusHelper from '../order-status-helper'

describe('OrderStatus', () => {
  const testProps = {
    orderId: orderStatus.OrderNumber,
    phoneNumber: '813-555-1234',
    err: null,
    loading: false,
    setOrderStatusState: jest.fn(),
    setOrderSearchState: jest.fn(),
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<OrderSearch { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when invalid phone number', () => {
    const tree = renderer.create(<OrderSearch { ...{ ...testProps, err: 'Invalid phone number.' } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when other error', () => {
    const tree = renderer.create(<OrderSearch { ...{ ...testProps, err: 'We are unable.' } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when loading', () => {
    const tree = renderer.create(<OrderSearch { ...{ ...testProps, loading: true } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('calls getRelatedOrders on find order click', () => {
    const tree = renderer.create(<OrderSearch { ...testProps } />)
    orderStatusHelper.getRelatedOrders = jest.fn()
    tree.root.findByProps({ className: 'primary' }).props.onClick()
    expect(orderStatusHelper.getRelatedOrders).toHaveBeenCalled()
  })

  it('calls setOrderSearchState on order number input change', () => {
    const tree = renderer.create(<OrderSearch { ...testProps } />)
    tree.root.findByProps({ id: 'order_status_number' }).props.onChange({ target: { value: 'test' } })
    expect(testProps.setOrderSearchState).toHaveBeenCalled()
  })

  it('calls setOrderSearchState on phone number input change', () => {
    const tree = renderer.create(<OrderSearch { ...testProps } />)
    tree.root.findByProps({ name: 'order_phone_number' }).props.onChange({ target: { value: 'test' } })
    expect(testProps.setOrderSearchState).toHaveBeenCalled()
  })
})
