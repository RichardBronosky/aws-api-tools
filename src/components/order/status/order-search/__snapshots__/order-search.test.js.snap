// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderStatus renders correctly when invalid phone number 1`] = `
<div
  className="grid-x order-search"
>
  <div
    className="cell small-12 medium-8 large-6 order-status-info"
  >
    <h1>
      View Order Status
    </h1>
    <p>
      Enter your order information to see your order details. You can pay an open balance, see your delivery date, and check your delivery timeframe. History is available for orders created within the last 180 days.
    </p>
    <form
      className="grid-x cell small-12"
    >
      <span
        className="error"
      >
        Invalid phone number.
      </span>
      <label
        className="cell small-12"
      >
        Order Number (ex: 12345678i)
      </label>
      <input
        className="cell small-12"
        id="order_status_number"
        maxLength={20}
        name="order_number"
        onChange={[Function]}
        type="text"
      />
      <label
        className="cell small-12"
      >
        Phone Number (ex: 8135551234)
      </label>
      <input
        className="cell small-12 error"
        name="order_phone_number"
        onChange={[Function]}
        type="tel"
      />
      <button
        className="primary"
        onClick={[Function]}
      >
        Find Order
      </button>
    </form>
  </div>
  <div
    className="cell small-12 medium-4 large-6 help-text"
  >
    <div
      className="background-box"
    >
      <h2>
        Need Help?
      </h2>
      <br />
      <p>
        Customer Care Center:
        <br />
        <a
          href="tel:(800) 766-6786"
        >
           (800) 766-6786
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`OrderStatus renders correctly when loading 1`] = `
<div
  className="grid-x order-search"
>
  <div
    className="cell small-12 medium-8 large-6 order-status-info"
  >
    <h1>
      View Order Status
    </h1>
    <p>
      Enter your order information to see your order details. You can pay an open balance, see your delivery date, and check your delivery timeframe. History is available for orders created within the last 180 days.
    </p>
    <form
      className="grid-x cell small-12"
    >
      <label
        className="cell small-12"
      >
        Order Number (ex: 12345678i)
      </label>
      <input
        className="cell small-12"
        id="order_status_number"
        maxLength={20}
        name="order_number"
        onChange={[Function]}
        type="text"
      />
      <label
        className="cell small-12"
      >
        Phone Number (ex: 8135551234)
      </label>
      <input
        className="cell small-12"
        name="order_phone_number"
        onChange={[Function]}
        type="tel"
      />
      <button
        className="primary"
        onClick={[Function]}
      >
        <img
          alt="loading matching orders"
          className="loader"
          src={Object {}}
        />
      </button>
    </form>
  </div>
  <div
    className="cell small-12 medium-4 large-6 help-text"
  >
    <div
      className="background-box"
    >
      <h2>
        Need Help?
      </h2>
      <br />
      <p>
        Customer Care Center:
        <br />
        <a
          href="tel:(800) 766-6786"
        >
           (800) 766-6786
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`OrderStatus renders correctly when other error 1`] = `
<div
  className="grid-x order-search"
>
  <div
    className="cell small-12 medium-8 large-6 order-status-info"
  >
    <h1>
      View Order Status
    </h1>
    <p>
      Enter your order information to see your order details. You can pay an open balance, see your delivery date, and check your delivery timeframe. History is available for orders created within the last 180 days.
    </p>
    <form
      className="grid-x cell small-12"
    >
      <span
        className="error"
      >
        We are unable.
      </span>
      <label
        className="cell small-12"
      >
        Order Number (ex: 12345678i)
      </label>
      <input
        className="cell small-12"
        id="order_status_number"
        maxLength={20}
        name="order_number"
        onChange={[Function]}
        type="text"
      />
      <label
        className="cell small-12"
      >
        Phone Number (ex: 8135551234)
      </label>
      <input
        className="cell small-12"
        name="order_phone_number"
        onChange={[Function]}
        type="tel"
      />
      <button
        className="primary"
        onClick={[Function]}
      >
        Find Order
      </button>
    </form>
  </div>
  <div
    className="cell small-12 medium-4 large-6 help-text"
  >
    <div
      className="background-box"
    >
      <h2>
        Need Help?
      </h2>
      <br />
      <p>
        Customer Care Center:
        <br />
        <a
          href="tel:(800) 766-6786"
        >
           (800) 766-6786
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`OrderStatus renders correctly with default props 1`] = `
<div
  className="grid-x order-search"
>
  <div
    className="cell small-12 medium-8 large-6 order-status-info"
  >
    <h1>
      View Order Status
    </h1>
    <p>
      Enter your order information to see your order details. You can pay an open balance, see your delivery date, and check your delivery timeframe. History is available for orders created within the last 180 days.
    </p>
    <form
      className="grid-x cell small-12"
    >
      <label
        className="cell small-12"
      >
        Order Number (ex: 12345678i)
      </label>
      <input
        className="cell small-12"
        id="order_status_number"
        maxLength={20}
        name="order_number"
        onChange={[Function]}
        type="text"
      />
      <label
        className="cell small-12"
      >
        Phone Number (ex: 8135551234)
      </label>
      <input
        className="cell small-12"
        name="order_phone_number"
        onChange={[Function]}
        type="tel"
      />
      <button
        className="primary"
        onClick={[Function]}
      >
        Find Order
      </button>
    </form>
  </div>
  <div
    className="cell small-12 medium-4 large-6 help-text"
  >
    <div
      className="background-box"
    >
      <h2>
        Need Help?
      </h2>
      <br />
      <p>
        Customer Care Center:
        <br />
        <a
          href="tel:(800) 766-6786"
        >
           (800) 766-6786
        </a>
      </p>
    </div>
  </div>
</div>
`;
